woocommerce/Settings_API/Abstract_Settings.php

```diff
@@ -35,6 +37,249 @@
  */
 abstract class Abstract_Settings {
 
+
+	/** @var string settings ID */
+	public $id = '';
+
+	/** @var Setting[] registered settings */
+	protected $settings = [];
+
+
+	/**
+	 * Constructs the class.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param string $id the ID of plugin or payment gateway that owns these settings
+	 */
+	public function __construct( $id ) {
+
+		$this->id = $id;
+
+		$this->register_settings();
+		$this->load_settings();
+	}
+
+
+	/**
+	 * Registers the settings.
+	 *
+	 * Plugins or payment gateways should overwrite this method to register their settings.
+	 *
+	 * @since x.y.z
+	 */
+	abstract protected function register_settings();
+
+
+	/**
+	 * Loads the values for all registered settings.
+	 *
+	 * @since x.y.z
+	 */
+	abstract protected function load_settings();
+
+
+	/**
+	 * Unregisters a setting.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param string $id setting ID to unregister
+	 */
+	public function unregister_setting( $id ) {
+
+		unset( $this->settings[ $id ] );
+	}
+
+
+	/**
+	 * Gets the settings ID.
+	 *
+	 * @since x.y.z
+	 *
+	 * @return string
+	 */
+	public function get_id() {
+
+		return $this->id;
+	}
+
+
+	/**
+	 * Gets all registered settings.
+	 *
+	 * @param array $ids setting IDs to get
```

Comment:
I believe we are trying to specify the array type when we can.
```suggestion
	 * @param string[] $ids setting IDs to get
```

Reply:
That makes sense. I committed your suggestions. Thanks!